<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>763</width>
    <height>465</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>IEEE754</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="5" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QLCDNumber" name="lcdFloatSign">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="statusTip">
         <string>Sign</string>
        </property>
        <property name="styleSheet">
         <string notr="true">background:#9a2;color: #fff</string>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Sunken</enum>
        </property>
        <property name="digitCount">
         <number>1</number>
        </property>
        <property name="mode">
         <enum>QLCDNumber::Bin</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLCDNumber" name="lcdFloatExponent">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="statusTip">
         <string>Exponent</string>
        </property>
        <property name="styleSheet">
         <string notr="true">background:#191;color: #fff</string>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Sunken</enum>
        </property>
        <property name="digitCount">
         <number>8</number>
        </property>
        <property name="mode">
         <enum>QLCDNumber::Bin</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLCDNumber" name="lcdFloatMantissa">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="statusTip">
         <string>Mantissa</string>
        </property>
        <property name="styleSheet">
         <string notr="true">background:#78c;color: #fff</string>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Sunken</enum>
        </property>
        <property name="digitCount">
         <number>23</number>
        </property>
        <property name="mode">
         <enum>QLCDNumber::Bin</enum>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="6" column="0">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>104</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="1" column="0">
     <widget class="QLineEdit" name="lineFloat">
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="maxLength">
       <number>127</number>
      </property>
     </widget>
    </item>
    <item row="14" column="0">
     <spacer name="verticalSpacer_2">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>106</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="7" column="0">
     <widget class="QLabel" name="labelDouble">
      <property name="text">
       <string>Double</string>
      </property>
     </widget>
    </item>
    <item row="10" column="0">
     <widget class="QLineEdit" name="lineDoubleHex">
      <property name="font">
       <font>
        <family>Consolas</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background: #444;color: #fff</string>
      </property>
      <property name="inputMask">
       <string>&gt;HH-HH-HH-HH-HH-HH-HH-HH;_</string>
      </property>
      <property name="text">
       <string>00-00-00-00-00-00-00-00</string>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QLineEdit" name="lineFloatStored">
      <property name="font">
       <font>
        <family>Consolas</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="statusTip">
       <string>Stored value</string>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="9" column="0">
     <widget class="QLineEdit" name="lineDoubleBin">
      <property name="font">
       <font>
        <family>Consolas</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background: #444;color: #fff</string>
      </property>
      <property name="inputMask">
       <string>BBBBBBBB-BBBBBBBB-BBBBBBBB-BBBBBBBB-BBBBBBBB-BBBBBBBB-BBBBBBBB-BBBBBBBB;_</string>
      </property>
      <property name="text">
       <string>00000000-00000000-00000000-00000000-00000000-00000000-00000000-00000000</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QLineEdit" name="lineFloatBin">
      <property name="font">
       <font>
        <family>Consolas</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background: #444;color: #fff</string>
      </property>
      <property name="inputMask">
       <string>BBBBBBBB-BBBBBBBB-BBBBBBBB-BBBBBBBB;_</string>
      </property>
      <property name="text">
       <string>00000000-00000000-00000000-00000000</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="labelFloat">
      <property name="text">
       <string>IEEE 754 Float</string>
      </property>
     </widget>
    </item>
    <item row="8" column="0">
     <widget class="QLineEdit" name="lineDouble">
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="maxLength">
       <number>127</number>
      </property>
     </widget>
    </item>
    <item row="12" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QLCDNumber" name="lcdDoubleSign">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="statusTip">
         <string>Sign</string>
        </property>
        <property name="styleSheet">
         <string notr="true">background:#9a2;color: #fff</string>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Sunken</enum>
        </property>
        <property name="digitCount">
         <number>1</number>
        </property>
        <property name="mode">
         <enum>QLCDNumber::Bin</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLCDNumber" name="lcdDoubleExponent">
        <property name="statusTip">
         <string>Exponent</string>
        </property>
        <property name="styleSheet">
         <string notr="true">background:#191;color: #fff</string>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Sunken</enum>
        </property>
        <property name="digitCount">
         <number>11</number>
        </property>
        <property name="mode">
         <enum>QLCDNumber::Bin</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLCDNumber" name="lcdDoubleMantissa">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="statusTip">
         <string>Mantissa</string>
        </property>
        <property name="styleSheet">
         <string notr="true">background:#78c;color: #fff</string>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Sunken</enum>
        </property>
        <property name="digitCount">
         <number>52</number>
        </property>
        <property name="mode">
         <enum>QLCDNumber::Bin</enum>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="3" column="0">
     <widget class="QLineEdit" name="lineFloatHex">
      <property name="font">
       <font>
        <family>Consolas</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background: #444;color: #fff</string>
      </property>
      <property name="inputMask">
       <string>&gt;HH-HH-HH-HH;_</string>
      </property>
      <property name="text">
       <string>00-00-00-00</string>
      </property>
     </widget>
    </item>
    <item row="11" column="0">
     <widget class="QLineEdit" name="lineDoubleStored">
      <property name="font">
       <font>
        <family>Consolas</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="statusTip">
       <string>Stored value</string>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>763</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <tabstops>
  <tabstop>lineFloat</tabstop>
  <tabstop>lineFloatBin</tabstop>
  <tabstop>lineFloatHex</tabstop>
  <tabstop>lineFloatStored</tabstop>
  <tabstop>lineDouble</tabstop>
  <tabstop>lineDoubleBin</tabstop>
  <tabstop>lineDoubleHex</tabstop>
  <tabstop>lineDoubleStored</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>lineFloat</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>floatChange(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>87</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>151</x>
     <y>168</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineFloatBin</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>floatChangeBin(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>249</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>301</x>
     <y>188</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineFloatHex</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>floatChangeHex(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>361</x>
     <y>132</y>
    </hint>
    <hint type="destinationlabel">
     <x>525</x>
     <y>181</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineDouble</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>doubleChange(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>166</x>
     <y>259</y>
    </hint>
    <hint type="destinationlabel">
     <x>135</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineDoubleBin</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>doubleChangeBin(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>251</x>
     <y>296</y>
    </hint>
    <hint type="destinationlabel">
     <x>274</x>
     <y>404</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineDoubleHex</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>doubleChangeHex(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>340</x>
     <y>320</y>
    </hint>
    <hint type="destinationlabel">
     <x>515</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>floatChange(QString)</slot>
  <slot>floatChangeBin(QString)</slot>
  <slot>floatChangeHex(QString)</slot>
  <slot>doubleChange(QString)</slot>
  <slot>doubleChangeBin(QString)</slot>
  <slot>doubleChangeHex(QString)</slot>
 </slots>
</ui>
